Tests/SwiftProtobufTests/Test_Conformance.swift

```diff
@@ -77,13 +77,57 @@ class Test_Conformance: XCTestCase, PBTestHelpers {
         }
     }
 
-    func testString_badUnicodeEscape() {
+    func testString_unicodeEscape() {
+        assertTextFormatDecodeSucceeds("optional_string: \"\\u1234\"") {
+            return $0.optionalString == "\u{1234}"
+        }
+        assertTextFormatDecodeSucceeds("optional_string: \"\\U0001F601\"") {
+            return $0.optionalString == "\u{1F601}"
+        }
+
+        assertTextFormatDecodeFails("optional_string: \"\\u")
+        assertTextFormatDecodeFails("optional_string: \"\\uDC\"")
+        assertTextFormatDecodeFails("optional_string: \"\\uDCXY\"")
+        assertTextFormatDecodeFails("optional_string: \"\\U")
+        assertTextFormatDecodeFails("optional_string: \"\\UDC\"")
+        assertTextFormatDecodeFails("optional_string: \"\\UDCXY\"")
+        assertTextFormatDecodeFails("optional_string: \"\\U1234DC\"")
+        assertTextFormatDecodeFails("optional_string: \"\\U1234DCXY\"")
```

Comment:
Be nice to verify that too-short escapes also fail to decode at the end of a string:

  " \u123"
  "\U1234567"

etc.

Reply:
Isn't that what `\UDC` and \`U1234DC` check?

Reply:
Oh.  Yeah.  Good point.